code-challenges.test.js

```diff
@@ -16,46 +23,144 @@
-
+describe("Converts", () => {
+  it("returns The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.", () => {
 var secretCodeWord1 = "Lackadaisical"
 // Expected output: "L4ck4d41s1c4l"
 var secretCodeWord2 = "Gobbledygook"
 // Expected output: "G0bbl3dyg00k"
 var secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
+expect(converts(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+expect(converts(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+expect(converts(secretCodeWord3)).toEqual("3cc3ntr1c")
 
-
+    })
+})
+// made a test function name "Converts"
+// my test statement is "returns The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0."
+// next i did 3 expect 
+// made the test fail
 
 // b) Create the function that makes the test pass.
+// function will be Converts into (str) aka string
+// thinking of using split becuase i want to remove certain letters in the string 
+// then use .delete becuase letters i am removing will be aeiou
+// and last one i will use 
+// i will use index
+
+const converts = (str) => {
+    var letters = str.split('')
+    for (let i=0; i<letters.length;i++){
+        if (letters[i].toLowerCase() === 'a') letters[i] = '4'
+        if (letters[i].toLowerCase() === 'e') letters[i] = '3'
+        if (letters[i].toLowerCase() === 'i') letters[i] = '1'
+        if (letters[i].toLowerCase() === 'o') letters[i] = '0'
+     
+    }
+    return letters.join('')
+ 
+ 
+
+}
+// since my first logic did not work i had some help in group study and change my original logic to a different one
+// so i have an input of a string
+// i split the string into an array of letters 
+// using this array i interate through to check each letter if they match the same
+// because i have to handle uppercase value as well i am calling .toLowerCase on each letter it doesn't matter if it's uppercase or lowercase
+// after i change the value we join it to a string 
+// and then i return the string 
+
+
+
+
+
+
 
 
 
 // --------------------2) Create a function that takes in an array and returns all the words that contain the letter a.
 
 // a) Create a test with an expect statement using the variable provided.
-
+describe("wordsContainA", () => {
+   test("returns all the words that contain the letter a", () => {
 var arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
 // Expected output: ["Apple", "Banana", "Orange"]
 var arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
 // Expected output: ["Mango", "Apricot", "Peach"]
+expect(wordscontaina(arrayOfWords1)).toEqual(["Apple", "Banana", "Orange"])
+expect(wordscontaina(arrayOfWords2)).toEqual(["Mango", "Apricot", "Peach"])
 
+   }) 
+})
+// made a test function name "wordsContainA"
+// my test returns all the words that contain the letter a.
+// next i did 2 expect 
+// made the test fail
 
 
 // b) Create the function that makes the test pass.
 
+const  wordscontaina = (arr) => {
+    var MatchLetterA = arr.filter((word) => {
```

Comment:
This is good, but remember we only use capital letters for the first word when we are dealing with "classes." The text editor changes the color for use so we know when we are looking at the right and wrong sorts of code.